### PR TITLE
add methods for writing data to temp file

### PR DESCRIPTION
**Pull request by ammarbinfaisal**
everything is almost done.

I've modified file and drectory methods to help in write and writeSync.
Just, the `filePath` argument which file function is accepting is getting appended to the the random path generated unlike what happens in temp-write, because using file method one can already specify the file name.

Or should I not make these changes to file and directory method. Just do all that in write and writeSync only? @sindresorhus 

**EDIT:** not asking this anymore

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ declare const tempy: {
 	/**
 	Get a temporary file path you can write to.
 
+	@param filePath - Path of the temporary file.
```
Comment (sindresorhus) [inline index.d.ts]: This should be an option, not a parameter, and it should have a name that better describes what it does.

The docs should also include the use-case for it, which you can find here: https://github.com/sindresorhus/temp-write/issues/2

File: index.d.ts [hunk 2]
```diff
@@ -55,7 +60,43 @@ declare const tempy: {
...
+
+	/**
+	Synchronously write string/buffer/stream to a random temp file.
+	
+	@param fileContent - Data to write to the temp file.
+	@param filePath - Path of the temporary file.
+	@param options - Options to be passed to tempy.file().
+	@returns The file path of the temp file.
+	@example
+	```
```
Comment (sindresorhus) [inline index.d.ts]: This is missing the import statement.

File: package.json [hunk 3]
```diff
@@ -35,9 +35,12 @@
 		"uniq"
 	],
 	"dependencies": {
+		"is-stream": "^2.0.0",
+		"make-dir": "^3.0.0",
 		"temp-dir": "^1.0.0",
 		"type-fest": "^0.3.1",
-		"unique-string": "^1.0.0"
+		"unique-string": "^1.0.0",
+		"util": "^0.12.0"
```
Comment (sindresorhus) [inline package.json]: `util` is a built-in core module.

File: test.js [hunk 4]
```diff
@@ -9,10 +10,26 @@ test('.file()', t => {
...
+	const tempPath = await tempy.write('rainbow');
+	t.deepEqual(fs.readFileSync(tempPath).toString(), 'rainbow');
+});
+
+test('.writeSync()', t => {
+	const tempPath = tempy.writeSync('rainbow', '/directo/ries', {name: 'custom-name.txt'});
+	t.deepEqual(fs.readFileSync(tempPath).toString(), 'rainbow');
+	t.true(tempPath.endsWith('custom-name.txt'));
+	t.true(tempPath.includes('/directo/ries'));
 });
```
Comment (sindresorhus) [inline test.js]: Needs a lot more tests. You're not even testing the buffer and stream input.

**Comment by sindresorhus** [comment]
From the issue:

> While doing that, look for ways to improve the original functionality and maybe add some more useful temp-related things.

I don't see this being done.

File: index.test-d.ts [hunk 5]
```diff
@@ -8,3 +8,5 @@ expectType<string>(tempy.file({extension: 'png'}));
 expectType<string>(tempy.file({name: 'afile.txt'}));
 expectError(tempy.file({extension: 'png', name: 'afile.txt'}));
 expectType<string>(tempy.root);
+expectType<string>(tempy.writeSync('hi there'));
+(async () => {expectType<string>(await tempy.write('hi there'))});
```
Comment (sindresorhus) [inline index.test-d.ts]: You can use top-level await here. No need for the async wrapper.

File: test.js (hunk 4 above)
Comment (ammarbinfaisal) [reply test.js]: apologies! I'll add them in a while.

**Comment by sindresorhus** [comment]
> > While doing that, look for ways to improve the original functionality and maybe add some more useful temp-related things.

> I don't see this being done.

I meant, propose things first adding it. Include what it would be useful for.

File: index.test-d.ts [hunk 6]
```diff
@@ -1,12 +1,14 @@
...
+    const options: tempy.Options = {};
+    expectType<string>(tempy.directory());
+    expectType<string>(tempy.file());
+    expectType<string>(tempy.file({extension: 'png'}));
+    expectType<string>(tempy.file({name: 'afile.txt'}));
+    expectError(tempy.file({extension: 'png', name: 'afile.txt'}));
+    expectType<string>(tempy.root);
+    expectType<string>(tempy.writeSync('hi there'));
+    expectType<string>(await tempy.write('hi there'));
+})
```
Comment (sindresorhus) [inline index.test-d.ts]: This is not top-level await. `await` is not at the top-level. You can drop the `(async () => {`, which is also invalid as it's missing `})` => `})()`.

**Comment by ammarbinfaisal** [comment]
That option to make the temp directory current working directory was inspired by https://pypi.org/project/temporary/ and I didnt think suffix, prefix and parent dir is required because there's already a way to provide name or path of the temporary file. exists() was taken from https://github.com/Flaque/filet

Arent these okay?

**Comment by ammarbinfaisal** [comment]
I had actually wrote this comment an hour ago but forgot to click comment button.

**Pull request closed without merging**